core/src/main/java/com/novoda/noplayer/mediaplayer/CheckBufferHeartbeatCallback.java

```diff
@@ -4,9 +4,7 @@
 import com.novoda.noplayer.Player;
 import com.novoda.noplayer.VideoPosition;
 
-@SuppressWarnings("PMD.RedundantFieldInitializer")
-        // we're being very explicit with our default field values, not a bad thing!
-class CheckBufferHeartbeatCallback implements Heart.Heartbeat.Callback<Player> {
+public class CheckBufferHeartbeatCallback implements Heart.Heartbeat.Callback<Player> {
```

Comment:
You are suggesting moving the `BufferListener` into `Heart.Heartbeat<T>`? I'm not sure it is the responsibility for the `Heart.Heartbeat` to know about this listener, it is only related to buffering hence there being a specific implementation for `bufferHeartbeat`. 

Perhaps I have not understood the benefit this will give us enough.